packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -65,6 +96,210 @@ describe('Sayt Component', () => {
     });
   });
 
+  describe('hideCorrectSayt()', () => {
+    it('should call hideSayt() when event specifies correct searchbox ID ', () => {
+      const hideSayt = stub(sayt, 'hideSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(true);
+      const event = {};
+
+      sayt.hideCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(hideSayt).to.be.calledOnce;
+    });
+
+    it('should not call hideSayt() when event specifies wrong searchbox ID', () => {
+      const hideSayt = stub(sayt, 'hideSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(false);
+      const event = {};
+
+      sayt.hideCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(hideSayt).to.not.be.called;
+    });
+  });
+
+  describe('isCorrectSayt()', () => {
+    it('should return true if event provides the correct searchbox ID', () => {
+      const searchbox = sayt.searchbox = 'some-searchbox-id';
+      const event = { detail: { searchbox } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return true if event does not provide a searchbox ID', () => {
+      sayt.searchbox = 'some-searchbox-id';
+      const event = { detail: {} };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return false if event provides the wrong searchbox ID', () => {
+      sayt.searchbox = 'correct-searchbox-id';
+      const event = { detail: { searchbox: 'wrong-searchbox-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true if event specifies a searchbox but SAYT has none specified', () => {
+      sayt.searchbox = undefined;
+      const event = { detail: { searchbox: 'some-searchbox-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should not error if passed an event without a detail attribute', () => {
+      const event = {};
+      const callback = () => sayt.isCorrectSayt(event);
+
+      expect(callback).to.not.throw();
+    });
+  });
+
+  describe('processClick()', () => {
+    let node: any = 'some-node';
+    let event: any;
+
+    beforeEach(() => {
+      event = { target: node };
+    });
+
+    it('should hide SAYT if the event target is nowhere relevant', () => {
+      stub(sayt, 'contains').returns(false);
+      stub(sayt, 'nodeInSearchBar').returns(false);
+      stub(sayt, 'hideSayt');
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
```

Comment:
Going back to the earlier discussion, this is actually an instance where we want to make assertions on `node` being the actual object since it's important that we're making assertions on the node itself.

```suggestion
      expect(sayt.contains).to.be.calledWith(sinon.match(node));
```

You will have to import `sinon` from the utils.

The same comment applies to all `contains` assertions below.

Reply:
As far as I can tell, `sinon.match()` does not match on identity:
```
sinon.match(object);
Requires the value to be not null or undefined and have at least the same properties as expectation.
```

Correct me if I'm wrong, but it doesn't sound like we gain anything from this change given that description.

Reply:
Oh, you're right, it should be `sinon.match.same()`. Hopefully I haven't been doing `.match` accidentally elsewhere where I should have used `.match.same` instead.